api/propagation/propagator.go

```diff
@@ -18,22 +18,47 @@ import (
 	"context"
 
 	"go.opentelemetry.io/api/core"
+	dctx "go.opentelemetry.io/api/distributedcontext"
 )
 
 // TextFormatPropagator is an interface that specifies methods to inject and extract SpanContext
-// into/from a carrier using Supplier interface.
+// and distributed context into/from a carrier using Supplier interface.
 // For example, HTTP Trace Context propagator would encode SpanContext into W3C Trace
 // Context Header and set the header into HttpRequest.
 type TextFormatPropagator interface {
 	// Inject method retrieves current SpanContext from the ctx, encodes it into propagator
 	// specific format and then injects the encoded SpanContext using supplier into a carrier
-	// associated with the supplier.
-	Inject(ctx context.Context, supplier Supplier)
+	// associated with the supplier. It also takes a correlationCtx whose values will be
+	// injected into a carrier using the supplier.
+	Inject(ctx context.Context, correlationCtx dctx.Map, supplier Supplier)
```

Comment:
dctx should already be in ctx. So Inject should simply retrieve it from the context.